zkevm-circuits/src/util/cell_manager_strategy.rs

```diff
@@ -0,0 +1,244 @@
+use std::collections::{BTreeMap, HashMap};
+
+use eth_types::Field;
+use halo2_proofs::plonk::{Advice, Column, ConstraintSystem};
+
+use super::cell_manager::{Cell, CellManagerColumns, CellManagerStrategy, CellType};
+
+#[derive(Clone, Debug, Default)]
+pub(crate) struct CMFixedWidthStrategyDistribution(HashMap<CellType, Vec<Column<Advice>>>);
+
+impl CMFixedWidthStrategyDistribution {
+    pub(crate) fn add(&mut self, cell_type: CellType, advice: Column<Advice>) {
+        if let Some(v) = self.0.get_mut(&cell_type) {
+            v.push(advice);
+        } else {
+            self.0.insert(cell_type, vec![advice]);
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct CMFixedWidthStrategy {
```

Comment:
I think this file should be documented via comments.  Maybe you can skip `CMFixedHeigthStrategy` because it's unfinished, but `CMFixedWidthStrategy` contains many things that I think should be documented like:
- What is `perm_substitution`?
- What is `next`?
- How does this strategy work?

Reply:
I have added doc comments.